static/js/message_edit.js

```diff
@@ -904,56 +906,57 @@ export function edit_last_sent_message() {
     });
 }
 
-function hide_delete_btn_show_spinner(deleting) {
-    if (deleting) {
-        $("do_delete_message_button").prop("disabled", true);
-        $("#delete_message_modal > div.modal-footer > button").hide();
-        const delete_spinner = $("#do_delete_message_spinner");
-        loading.make_indicator(delete_spinner, {abs_positioned: true});
-    } else {
-        loading.destroy_indicator($("#do_delete_message_spinner"));
-        $("#do_delete_message_button").prop("disabled", false);
-        $("#delete_message_modal > div.modal-footer > button").show();
-    }
-}
-
 export function delete_message(msg_id) {
     $("#delete-message-error").html("");
-    overlays.open_modal("#delete_message_modal");
+
+    const html_body = render_delete_message_modal_body();
+    const modal_parent = $("#message_feed_container");
+
+    function do_delete_message() {
+        confirm_dialog.show_confirm_dialog_spinner();
+
+        currently_deleting_messages.push(msg_id);
+        channel.del({
+            url: "/json/messages/" + msg_id,
+            success() {
+                currently_deleting_messages = currently_deleting_messages.filter(
+                    (id) => id !== msg_id,
+                );
+                confirm_dialog.hide_confirm_dialog_spinner();
+                overlays.close_modal("#confirm_dialog_modal");
+            },
+            error(xhr) {
+                currently_deleting_messages = currently_deleting_messages.filter(
+                    (id) => id !== msg_id,
+                );
+
+                confirm_dialog.hide_confirm_dialog_spinner();
+                overlays.close_modal("#confirm_dialog_modal");
+                ui_report.error(
+                    $t_html({defaultMessage: "Error deleting message"}),
+                    xhr,
+                    $("#delete-message-error"),
```

Comment:
Where does this error appear?  If it's in the modal, we shouldn't be closing it.